### ISSUE TITLE
failing test in com.ofg.uptodate.finder.AllRepositoriesnewVersionFinderSpec

### ISSUE THREAD

**Issue by devopsd** [4finance/uptodate-gradle-plugin#30]
`should list all dependencies, that have newer versions` test fails with the following error:
```
Too few invocations for:

1 * loggerProxy.warn(_, "New versions available:\n" +
                    "'junit:junit:4.11'\n" +
                    "'org.hibernate:hibernate-core:4.3.6.Final'")   (0 invocations)

Unmatched invocations (ordered by similarity):

1 * loggerProxy.warn(Logger[com.ofg.uptodate.finder.NewVersionFinderInAllRepositories], 'New versions available:\n\'org.hibernate:hibernate-core:4.3.6.Final\'\n\'junit:junit:4.11\'')
1 * loggerProxy.debug(Logger[com.ofg.uptodate.finder.NewerDependencyProvider], 'Dependencies to get only newer: [\'org.hibernate:hibernate-core:4.2.9.Final\', \'org.hibernate:hibernate-core:4.3.6.Final\']')
2 * loggerProxy.debug(Logger[com.ofg.uptodate.finder.NewerDependencyProvider], 'Fetched dependency has greater version than the current one [false]')
1 * loggerProxy.debug(Logger[com.ofg.uptodate.finder.NewerDependencyProvider], 'Dependencies to get only newer: [\'junit:junit:4.10\', \'junit:junit:3.7\']')
1 * loggerProxy.debug(Logger[com.ofg.uptodate.finder.NewVersionFinderInAllRepositories], 'Unique newest versions found [\'org.hibernate:hibernate-core:4.3.6.Final\', \'junit:junit:4.11\']')
1 * loggerProxy.debug(Logger[com.ofg.uptodate.finder.NewVersionFinder], 'Newer dependencies found in http://localhost:12306/maven/: [\'junit:junit:4.11\']')
1 * loggerProxy.debug(Logger[com.ofg.uptodate.finder.NewerDependencyProvider], 'Dependencies to get only newer: [\'org.hibernate:hibernate-core:4.2.9.Final\', \'org.hibernate:hibernate-core:2.0.0.Final\']')
1 * loggerProxy.debug(Logger[com.ofg.uptodate.finder.NewVersionFinder], 'Newer dependencies found in http://localhost:12306/jcenter/: [\'org.hibernate:hibernate-core:4.3.6.Final\']')
1 * loggerProxy.debug(Logger[com.ofg.uptodate.finder.NewerDependencyProvider], 'Dependencies to get only newer: [\'junit:junit:4.10\', \'junit:junit:4.11\']')
2 * loggerProxy.debug(Logger[com.ofg.uptodate.finder.NewerDependencyProvider], 'Fetched dependency has greater version than the current one [true]')
```

**Comment by szpak** [comment]
@devopsd What OS and Gradle do you use?

@cbuleandra you reverse it recently in 153190d9ba45c38786f1a240ae40f22ca9fe23c3. Is that order constant?

**Comment by devopsd** [comment]
Fedora 20
Gradle 1.12
Java 1.8.0_31
IntelliJ IDEA 14.1

**Comment by devopsd** [comment]
It's funny, when executed from console it passes, but from IDEA it fails

**Comment by cbuleandra** [comment]
@szpak I thought it would be constant, but I missed the collection being transformed to a HashSet here: https://github.com/4finance/uptodate-gradle-plugin/blob/master/src/main/groovy/com/ofg/uptodate/finder/NewVersionFinderInAllRepositories.groovy#L25
We can either change that to explicitly create a LinkedHashSet instead and ensure the insertion order is maintained, or we could update the way we do the verifications in the tests to not care about the order of the dependencies. Personally, I would suggest going with the second approach, since the order we list them does not seem important to me.
What do you think?

**Comment by szpak** [comment]
I wonder If the received list is always in specific order (e.g. alphabetical). If yes maybe it would be good to preserve it. If even not, maybe it would be a good feature to always report available updates sorted. Then all Spring updates would be presented together, etc.

WDYT guys?

**Comment by kamilszymanski** [comment]
@devopsd does it also fail with fixes from #32?

**Comment by devopsd** [comment]
Confirming, #32 solves the issue

**Comment by kamilszymanski** [comment]
Fixed by #32
